packages/flutter_tools/test/general.shard/intellij/intellij_validator_test.dart

```diff
@@ -496,3 +519,31 @@ void createIntellijDartPluginJar(String pluginJarPath, FileSystem fileSystem) {
     ..createSync(recursive: true)
     ..writeAsBytesSync(ZipEncoder().encode(dartPlugins)!);
 }
+
+// TODO(fujino): we should use the MemoryFileSystem and a FileExceptionHandler,
+// but we will need to teach package:file about the Directory.listSync
+// operation first.
```

Comment:
I'm confused. Are we unable to use `FileExceptionHandler` because the `Directory.listSync` method in `path` doesn't hookup `opHandle`? Said another way, I don't understand what "teaching" `package:file` about `Directory.listSync` means. Could you clarify?

Reply:
> Are we unable to use FileExceptionHandler because the Directory.listSync method in path doesn't hookup opHandle?

Exactly. If you like `MemoryFileSystemEntity.internalDeleteSync()`, it will call `fileSystem.opHandle()` specifically for the sake of tests that want to inject a callback: https://github.com/google/file.dart/blob/master/packages/file/lib/src/backends/memory/memory_file_system_entity.dart#L291

We would have to do something similar in `MemoryDirectory.listSync()`: https://github.com/google/file.dart/blob/master/packages/file/lib/src/backends/memory/memory_directory.dart#L127

Reply:
Cool. My instinct says that we should file a feature request against package:file and then update this code comment to be a little less vague while also including a link to the feature request. This would save future readers from similar confusion. However, I understand doing all of this might not be worth the time.

Reply:
done: https://github.com/google/file.dart/issues/227